include/benchmark/benchmark_api.h

```diff
@@ -429,6 +553,33 @@ class State {
   BENCHMARK_ALWAYS_INLINE
   size_t iterations() const { return total_iterations_; }
 
+
+  // Add a user counter. Return the counter handle.
+  BENCHMARK_ALWAYS_INLINE
+  size_t SetCounter(const char* n) { return counters_.Set(n); }
```

Comment:
given these always return a handle, i see no reason to allow counter lookup by name. it's just going to cause developers pain in the future when their benchmarks take longer than they expected.

Please simplify this such that SetCounter (actually, CreateCounter) takes a string and everything else takes a handle.